src/QMCDrivers/DMC/WalkerControl.cpp

```diff
@@ -517,7 +529,14 @@ void WalkerControl::swapWalkersSimple(MCPopulation& pop)
   for (int iw = 0; iw < good_walkers.size(); iw++)
     TotalMultiplicity += good_walkers[iw]->Multiplicity;
   if (static_cast<int>(TotalMultiplicity) != fair_offset_[rank_num_ + 1] - fair_offset_[rank_num_])
-    throw std::runtime_error("Multiplicity check failed in WalkerControl::swapWalkersSimple!");
+  {
+    std::ostringstream error_msg;
+    error_msg << "Multiplicity check failed in WalkerControl::swapWalkersSimple!\n"
+              << "for rank: " << rank_num_ << " total_multiplicity: " << TotalMultiplicity
+              << "  fair_offset_[rank_num_ + 1] - fair_offset_[rank_num_]: " << fair_offset_[rank_num_ + 1] << " - "
+              << fair_offset_[rank_num_] << '\n';
+    throw std::runtime_error(error_msg.str());
```

Comment:
This is just a less useless error message.  It's in an 
```
#ifndef NDEBUG
```
section so this has no effect on production science